src/Streamly/Internal/Data/Unfold.hs

```diff
@@ -356,17 +360,31 @@ discardSecond = lmap fst
 swap :: Unfold m (a, c) b -> Unfold m (c, a) b
 swap = lmap Tuple.swap
 
+-- See StreamD.scanlM' for implementing this.
+--
+-- | Scan the input of an 'Unfold' to change it in a stateful.
+--
+-- /Unimplemented/
+{-# INLINE_NORMAL lscanlM' #-}
+lscanlM' :: -- Monad m =>
+    (b -> a -> m b) -> m b -> Unfold m b c -> Unfold m a c
+lscanlM' = undefined
+
 -------------------------------------------------------------------------------
 -- Output operations
 -------------------------------------------------------------------------------
 
+-- XXX Do we need this combinator or the stream based idiom is enough?
```

Comment:
This is a nice combinator. We can have this for completeness.

Reply:
We can change the implementation to stream based idiom, if the perf is not affected.